lib/bap_image/objectFile.mli

```diff
@@ -0,0 +1,50 @@
+(** LLVM object file *)
+
+open Core_kernel.Std
+
+type t
+
+external create : Bigstring.t -> t Or_error.t =
+  "object_file_create_stub"
+
+type archType =
```

Comment:
btw, if you don't want to mess with creating polymorphic variants from C (it is not hard, anyway) you can just use strings. BAP's `Arch.of_string` understands all LLVM architectures.